server-spi-private/src/main/java/org/keycloak/models/utils/KeycloakModelUtils.java

```diff
@@ -318,17 +319,20 @@ public static ComponentModel getComponentModel(KeycloakSessionFactory factory, S
         private final String componentId;
         private final String providerId;
         private final String providerType;
+        private final String providerProfile;
         private final String realmId;
         private final Scope config;
 
-        public ScopeComponentModel(Class<?> providerClass, Scope baseConfiguration, String spiName, String realmId) {
+        public ScopeComponentModel(final Class<?> providerClass, final Scope baseConfiguration, final String spiName, final String realmId,
+                                   final String modelName) {
             final String pr = baseConfiguration.get("provider", Config.getProvider(spiName));
-
             this.providerId = pr == null ? "default" : pr;
+            this.providerType = providerClass.getName();
+            this.providerProfile = getProviderProfile(baseConfiguration, spiName, providerId);
             this.config = baseConfiguration.scope(this.providerId);
-            this.componentId = spiName + "- " + (realmId == null ? "" : "f:" + realmId + ":") + this.providerId;
+            this.componentId = spiName + "-" + (realmId == null ? "" : "f:" + realmId + ":") + this.providerId + ":" +
+                    (this.config.isEmpty()? this.providerProfile : modelName + "-inline");
```

Comment:
If the "inline" config is not empty, we assume it is trying to override one of the profiles with some area-specific needs, and thus we craft a componentId that conveys that.